src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -191,11 +194,14 @@ protected void buildEndpoint(ApiEndpoint baseEndpoint) {
         int offset = HttpUtils.getQueryParamFromRequest(req, OFFSET_PARAM, 0, DEFAULT_OFFSET);
         String userId = HttpUtils.getQueryParamFromRequest(req, USER_ID_PARAM, true);
         // Filter the response based on the user id, if provided.
-        if (userId != null) {
-            return persistence.getResponseList(Filters.eq(USER_ID_PARAM, userId), offset, limit);
-        }
         // If the user id is not provided filter response based on requesting user.
         RequestingUser requestingUser = Auth0Connection.getUserFromRequest(req);
+        if (userId != null) {
+            OtpUser otpUser = Persistence.otpUsers.getById(userId);
+            return (otpUser != null && otpUser.canBeManagedBy(requestingUser)) ?
```

Comment:
If the OtpUser cannot be managed by the requesting user, as written, this returns 200-OK (and a `null` body).
This should return a 403 error instead.

Reply:
Updated.